c2/c2-client-bundle/c2-client-service/src/main/java/org/apache/nifi/c2/client/service/FlowUpdateInfoHolder.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.nifi.c2.client.service;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.Optional;
+import java.util.Properties;
+import org.apache.nifi.c2.client.api.FlowUpdateInfo;
+import org.apache.nifi.util.file.FileUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class FlowUpdateInfoHolder {
+    private static final Logger LOGGER = LoggerFactory.getLogger(FlowUpdateInfoHolder.class);
+    private static final String FLOW_IDENTIFIER_FILENAME = "flow-identifier";
+    private static final String C2_FLOW_URL_KEY = "c2.flow.url";
```

Comment:
I was thinking about this, do we need to persist the url? I think you could have just the file named flow-identifier with one value in it the actual flow id. That way this would become simpler no need for the properties class and property keys at all.
This also has the implication that I don't think we need to store the url in FlowUpdateInfo either. The url is used only in one place and at that point we have the incoming operation with the location attribute which is exactly this.
What do you think? @mattyb149 would you be ok with removing the url from FlowUpdateInfo as it is not needed there based on the current code structure.

Reply:
Yes, you are right, the URL is never used after we persist the config.yml. I think we can safely remove it.